### PR TITLE
[vcpkg_from_gitlab] improvements 

### PR DESCRIPTION
**Pull request by autoantwort**
For `vcpkg_from_github`: Correctly set `VCPKG_HEAD_VERSION` (Used for the `Version` field in the `CONTROL` in the `packages` folder, but this field is used nowhere).
For `vcpkg_from_gitlab`: Use the gitlab api instead of `git ls-remote`.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Sorry for late, can you please resolve the file conflicts first?

Thanks.

**Comment by JackBoosY** [comment]
Will review this today.

**Comment by JackBoosY** [comment]
The `vcpkg_download_distfile` changes is good, but I lack knowledge of github api and hope other team members review this PR.

**Comment by JackBoosY** [comment]
Tagged `reviewed` so other members will review this changes (doesn't means approved).

File: scripts/cmake/vcpkg_from_gitlab.cmake [hunk 1]
```diff
@@ -69,6 +69,21 @@ This exports the `VCPKG_HEAD_VERSION` variable during head builds.
 
 include(vcpkg_execute_in_download_mode)
 
+function(z_uri_encode input outpu_variable)
```
Comment (Thomas1664) [inline scripts/cmake/vcpkg_from_gitlab.cmake:72]:
```suggestion
function(z_uri_encode input output_variable)
```

File: scripts/cmake/vcpkg_from_gitlab.cmake [hunk 2]
```diff
@@ -69,6 +69,21 @@ This exports the `VCPKG_HEAD_VERSION` variable during head builds.
...
+    math(EXPR last "${length} - 1")
+    set(result "")
+    foreach(i RANGE ${last})
+        math(EXPR even "${i} % 2")
+        if("${even}" STREQUAL "0")
+            string(SUBSTRING "${hex}" "${i}" 2 char)
+            string(APPEND result "%${char}")
+        endif()
+    endforeach()
+    set("${outpu_variable}" ${result} PARENT_SCOPE)
```
Comment (Thomas1664) [inline scripts/cmake/vcpkg_from_gitlab.cmake:84]:
```suggestion
    set("${output_variable}" ${result} PARENT_SCOPE)
```

**Comment by Cheney-W** [comment]
@BillyONeal Could you please help review this PR? Thank you!

**Comment by BillyONeal** [comment]
Tested and this worked with atk. Thanks!

**Pull request merged**